src/overlays/actors/ovl_Obj_Kibako2/z_obj_kibako2.c

```diff
@@ -22,50 +30,233 @@ const ActorInit Obj_Kibako2_InitVars = {
+        if (skulltula != NULL) {
+            skulltula->parent = &this->dyna.actor;
+            skulltula->velocity.y = 13.0f;
+            skulltula->speedXZ = 0.0f;
+        }
+    }
+}
 
-extern UNK_TYPE D_06000960;
-extern UNK_TYPE D_06000B70;
-extern UNK_TYPE D_06001040;
+void ObjKibako2_SpawnContents(ObjKibako2* this, GlobalContext* globalCtx) {
+    if (KIBAKO2_CONTENTS(&this->dyna.actor) == CONTENTS_COLLECTIBLE) {
+        ObjKibako2_SpawnCollectible(this, globalCtx);
+    } else {
+        ObjKibako2_SpawnSkulltula(this, globalCtx);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Kibako2/func_8098E5C0.s")
+void ObjKibako2_Init(Actor* thisx, GlobalContext* globalCtx) {
+    ObjKibako2* this = THIS;
+    s32 pad;
+    ObjKibako2Contents contents;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Kibako2/func_8098E62C.s")
+    contents = KIBAKO2_CONTENTS(&this->dyna.actor);
+    BcCheck3_BgActorInit(&this->dyna, 0);
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    BgCheck3_LoadMesh(globalCtx, &this->dyna, &D_06000B70);
+    Collider_SetCylinder(globalCtx, &this->collider, &this->dyna.actor, &sCylinderInit);
+    Collider_UpdateCylinder(&this->dyna.actor, &this->collider);
+    this->dyna.actor.home.rot.z = 0;
+    this->dyna.actor.world.rot.z = 0;
+    this->dyna.actor.shape.rot.z = 0;
+    this->dyna.actor.world.rot.x = 0;
+    this->dyna.actor.shape.rot.x = 0;
+    if (contents == CONTENTS_COLLECTIBLE) {
+        if (func_800A81A4(globalCtx, KIBAKO2_COLLECTIBLE_ID(&this->dyna.actor),
+                          KIBAKO2_COLLECTIBLE_FLAG(&this->dyna.actor))) {
+            this->unk_1AC = 1;
+            this->dyna.actor.flags |= 0x10;
+        }
+    }
+    if ((contents != CONTENTS_SKULLTULA) || (!ObjKibako2_ContainsSkulltula(this, globalCtx))) {
+        this->skulltulaNoiseTimer = -1;
+    }
+    this->actionFunc = ObjKibako2_Idle;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Kibako2/func_8098E8A8.s")
+void ObjKibako2_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    ObjKibako2* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Kibako2/func_8098E900.s")
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+    BgCheck_RemoveActorMesh(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Kibako2/func_8098E9C4.s")
+s32 ObjKibako2_ShouldBreak(ObjKibako2* this) {
+    u8 acFlags = this->collider.base.acFlags;
+    s32 shouldBreak = false;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Kibako2/ObjKibako2_Init.s")
+    if ((this->collider.base.acFlags & AC_HIT)) {
+        Actor* ac = this->collider.base.ac;
+        this->collider.base.acFlags = acFlags & ~AC_HIT;
+        if (ac != NULL) {
+            if (this->collider.info.acHitInfo->toucher.dmgFlags & 1 << 31) {
```

Comment:
Ah yeah, I see the need for the parentheses because it's actually `dmgFlags & (1 << 31)`, not `(dmgFlags & 1) << 31`. I'll add them, though the Goron Punch/Pound case already has them.